src/number_theory/number_field/embeddings.lean

```diff
@@ -123,3 +125,124 @@ end
+
+lemma place_conjugate (φ : K →+* ℂ) : place (conjugate φ) = place φ :=
+by { ext, simp only [place_apply, norm_eq_abs, abs_conj, conjugate_coe_eq] }
+
+/-- A embedding into `ℂ` is real if it is fixed by complex conjugation. -/
+@[reducible] def is_real (φ : K →+* ℂ) : Prop := is_self_adjoint φ
+
+lemma is_real_iff {φ : K →+* ℂ} : is_real φ ↔ conjugate φ = φ := is_self_adjoint_iff
+
+/-- A real embedding as a ring homomorphism from `K` to `ℝ` . -/
+def is_real.embedding {φ : K →+* ℂ} (hφ : is_real φ) : K →+* ℝ :=
+{ to_fun := λ x, (φ x).re,
+  map_one' := by simp only [map_one, one_re],
+  map_mul' := by simp only [complex.eq_conj_iff_im.mp (ring_hom.congr_fun hφ _), map_mul, mul_re,
+  mul_zero, tsub_zero, eq_self_iff_true, forall_const],
+  map_zero' := by simp only [map_zero, zero_re],
+  map_add' := by simp only [map_add, add_re, eq_self_iff_true, forall_const], }
+
+@[simp]
+lemma is_real.coe_embedding_apply {φ : K →+* ℂ} (hφ : is_real φ) (x : K) :
+  (hφ.embedding x : ℂ) = φ x :=
+begin
+  ext, { refl, },
+  { rw [of_real_im, eq_comm, ← complex.eq_conj_iff_im],
+    rw is_real at hφ,
+    exact ring_hom.congr_fun hφ x, },
+end
+
+lemma place_real_embedding_eq_place {φ : K →+* ℂ} (hφ : is_real φ) :
+  place hφ.embedding = place φ :=
+by { ext x, simp only [place_apply, real.norm_eq_abs, ←abs_of_real, norm_eq_abs,
+  real_embedding_eq_embedding hφ x], }
+
+lemma is_real_conjugate_iff {φ : K →+* ℂ} :
+  is_real (conjugate φ) ↔ is_real φ := is_self_adjoint.star_iff
+
+end number_field.complex_embedding
+
+section infinite_place
+
+open number_field
+
+variables (K : Type*) [field K]
+
+/-- An infinite place of a number field `K` is a place associated to a complex embedding. -/
+def number_field.infinite_place := { w : absolute_value K ℝ  // ∃ φ : K →+* ℂ, place φ = w}
+
+instance [number_field K] : nonempty (number_field.infinite_place K) :=
+begin
+  rsuffices ⟨φ⟩ : nonempty (K →+* ℂ), { use ⟨place φ, ⟨φ, rfl⟩⟩, },
+  rw [← fintype.card_pos_iff, embeddings.card K ℂ],
+  exact finite_dimensional.finrank_pos,
+end
+
+variables {K}
+
+/-- Return the infinite place defined by a complex embedding `φ`. -/
+noncomputable def number_field.infinite_place.mk (φ : K →+* ℂ) : number_field.infinite_place K :=
+⟨place φ, ⟨φ, rfl⟩⟩
+
+namespace number_field.infinite_place
+
+open number_field
+
+instance : has_coe_to_fun (infinite_place K) (λ _, K → ℝ) := { coe := λ w, w.1 }
+
+instance : monoid_with_zero_hom_class (infinite_place K) K ℝ :=
+{ coe := λ w x, w.1 x,
+  coe_injective' := λ _ _ h, subtype.eq (absolute_value.ext (λ x, congr_fun h x)),
+  map_mul := λ w _ _, w.1.map_mul _ _,
+  map_one := λ w, w.1.map_one,
+  map_zero := λ w, w.1.map_zero, }
+
+instance : nonneg_hom_class (infinite_place K) K ℝ :=
+{ coe :=  λ w x, w x,
+  coe_injective' := λ _ _ h, subtype.eq (absolute_value.ext (λ x, congr_fun h x)),
+  map_nonneg := λ w x, w.1.nonneg _ }
+
+lemma coe_mk (φ : K →+* ℂ) : mk φ = place φ := rfl
```

Comment:
Ah, you need to write the coe:
```suggestion
lemma coe_mk (φ : K →+* ℂ) : ⇑(mk φ) = place φ := rfl
```

Reply:
This doesn't work since there is not lift from `infinite_place K` to `absolute_value K R` (Should there be one?). Otherwise, this works: 
```lean
lemma coe_mk (φ : K →+* ℂ) : (mk φ).1 = place φ := rfl
```